meson_init.sh

```diff
@@ -105,6 +106,7 @@ for platform in ${PLATFORMS[@]}; do
   mkdir -p "$obj_dir"
   meson ${FLAGS_reconfigure} \
     -Dtarget="$platform" \
+    -Dot_version="$OT_VERSION" \
```

Comment:
nit: I see a mix of `${FOO}` and `$FOO`.  I'd be nice to stick to one approach. Marking as nit since we don't have a .sh style guide I believe?

Reply:
*shrug*

I think we probably want to shell style guide at some point, so I wouldn't bother picking one over the other before we go through that bikeshed.